### PR TITLE
Updated cert-manager version v1.11.0---> v1.15.1   

### PR DESCRIPTION
**Pull request by Horiodino**
**Please specify the area for this PR**

**What does does this PR do / why we need it**:
cert-manager version v1.11.0---> v1.13.1 (latest version)

**Which issue(s) this PR fixes**:
Fixes # https://github.com/devfile/api/issues/1320

**PR acceptance criteria**:

- [✅  ] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [ ] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the registry operator documentation need to be updated with your changes?

**Testing changes**
```
kubectl get pods -n cert-manager
NAME                                       READY   STATUS    RESTARTS   AGE
cert-manager-67f7b4f67d-jc9dx              1/1     Running   0          2m1s
cert-manager-cainjector-564b978556-kdsnw   1/1     Running   0          2m1s
cert-manager-webhook-7d54ddf58b-ldntk      1/1     Running   0          2m1s
```

[Running Unit Tests](https://github.com/devfile/registry-operator/blob/main/CONTRIBUTING.md#unit-tests)

[Running Integration Tests](https://github.com/devfile/registry-operator/blob/main/CONTRIBUTING.md#integration-tests)

**Special notes to the reviewer**:

### CODE REVIEW

**Review by Jdubrick — COMMENTED** [review]
/ok-to-test

File: Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@
 VERSION ?= `cat $(PWD)/VERSION`
 # Default bundle image tag
 BUNDLE_IMG ?= quay.io/devfile/registry-operator-bundle:v$(VERSION)
-CERT_MANAGER_VERSION ?= v1.11.0
+CERT_MANAGER_VERSION ?= v1.13.1
```
Comment (Jdubrick) [inline Makefile:20]: Where are we finding `v1.13.1` as the latest? I am looking here https://cert-manager.io/docs/releases/ and see `1.15` as the latest with `1.12` as LTS.

Do we want the latest or something with LTS? @michael-valdron @thepetk

Comment (thepetk) [reply Makefile:20]: @Jdubrick I think is a fair point. I'd suggest the latest version that doesn't introduce breaking changes affecting our operateor. Looking at the same link you've shared I see that the suggested version `1.13.1` is no longer supported. As a result, if possible I'd suggest going for >1.14 version as they are supported.

**Review by thepetk — COMMENTED** [review]
replied on @Jdubrick comment

**Comment by Horiodino** [comment]
these days i always get google ai based search results that's why i use `1.13`   , thanks for the pointing it out

**Comment by Horiodino** [comment]
```
make install-cert
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.1/cert-manager.yaml
namespace/cert-manager created
customresourcedefinition.apiextensions.k8s.io/certificaterequests.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/certificates.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/challenges.acme.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/clusterissuers.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/issuers.cert-manager.io created
customresourcedefinition.apiextensions.k8s.io/orders.acme.cert-manager.io created
```

```
kubectl get pods -n cert-manager
NAME                                      READY   STATUS    RESTARTS   AGE
cert-manager-cainjector-9d956987c-fwpkc   1/1     Running   0          59s
cert-manager-fdd97855b-zk8ff              1/1     Running   0          59s
cert-manager-webhook-9f799c7d7-wql9s      1/1     Running   0          59s
```

**Review by michael-valdron — CHANGES_REQUESTED** [review]
We'll need to update this in the documentation as well: https://github.com/devfile/registry-operator/blob/eb1219b52113fc95161e9ea6578adb1b8f0fb988/README.md?plain=1#L82

**Review by thepetk — APPROVED** [review]
/lgtm after all checks are passing

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**